src/object/cli_shard.c

```diff
@@ -1864,15 +1869,21 @@ obj_shard_query_recx_post(struct obj_query_key_cb_args *cb_args, uint32_t shard,
 	end[0] = DAOS_RECX_END(recx[0]);
 	end[1] = DAOS_RECX_END(recx[1]);
 	if (get_max) {
-		if (end[0] > end[1])
-			*result_recx = recx[0];
-		else
-			*result_recx = recx[1];
+		if (end[0] > end[1]) {
+			if (DAOS_RECX_END(*result_recx) < end[0] || changed)
+				*result_recx = recx[0];
+		} else {
+			if (DAOS_RECX_END(*result_recx) < end[1] || changed)
+				*result_recx = recx[1];
+		}
 	} else {
-		if (end[0] < end[1])
-			*result_recx = recx[0];
-		else
-			*result_recx = recx[1];
+		if (end[0] < end[1]) {
+			if (DAOS_RECX_END(*result_recx) > end[0] || changed)
+				*result_recx = recx[0];
+		} else {
+			if (DAOS_RECX_END(*result_recx) > end[0] || changed)
```

Comment:
here should be "end[0]" or "end[1]"?

Reply:
ah, yes, sure,  I will fix it.  Sorry, only get_max is used for the moment. I will fix this in PR-5924.